src/server.cpp

```diff
@@ -139,7 +139,60 @@ v3f ServerSoundParams::getPos(ServerEnvironment *env, bool *pos_exists) const
 	return v3f(0,0,0);
 }
 
+void Server::ShutdownState::reset()
+{
+	m_timer = 0.0f;
+	message.clear();
+	should_reconnect = false;
+	is_requested = false;
+}
+
+void Server::ShutdownState::trigger(float delay, const std::string &msg, bool reconnect)
+{
+	m_timer = delay;
+	message = msg;
+	should_reconnect = reconnect;
+}
 
+void Server::ShutdownState::tick(float dtime, Server *server)
+{
+	// Timed shutdown
+	static const float shutdown_msg_times[] =
+	{
+		1, 2, 3, 4, 5, 10, 15, 20, 25, 30, 45, 60, 120, 180, 300, 600, 1200, 1800, 3600
```

Comment:
Perhaps we should re-consider the notification times? 1 second can almost not be read, unless there are network delays - and `?5` seem way to detailed. If we continue the scheme `/2` `*2/3` scheme from the end of this list, we would get an array like this: (discontinued after `20`, where it would be an unhandy `13.333` seconds)
`1, 2, 3, 4, 5, 10, 20, 40, 60, 120, 180, 300, 600, 1200, 1800, 3600`

Reply:
1 can be read :) i was based on WoW experience for those messages